install_config/provisioners.adoc

```diff
@@ -0,0 +1,198 @@
+
+.Required EFS Ansible Variables
+[options="header"]
+|===
+
+|Variable |Description
+
+|`openshift_provisioners_efs_fsid`
+|The link:http://docs.aws.amazon.com/efs/latest/ug/gs-step-two-create-efs-resources.html[File system ID] of the EFS file system, e.g. fs-47a2c22e
+
+|`openshift_provisioners_efs_region`
+|The Amazon EC2 region of the EFS file system.
+
+|`openshift_provisioners_efs_aws_access_key_id`
+|The AWS access key of the IAM user, used to check that the EFS file system specified actually exists.
+
+|`openshift_provisioners_efs_aws_secret_access_key`
+|The AWS secret access key of the IAM user, used to check that the EFS file system specified actually exists.
+
+|===
+
+.Optional EFS Ansible Variables
+[options="header"]
+|===
+
+|Variable |Description
+
+|`openshift_provisioners_efs`
+| When `True` the AWS EFS provisioner will be installed or uninstalled according to whether `openshift_provisioners_install_provisioners` is `True` or `False`, respectively. Defaults to `False`.
+
+|`openshift_provisioners_efs_path`
+| The path of the directory in the EFS file system in which the EFS provisioner will create a directory to back each PV it creates. It must exist and be mountable by the EFS provisioner. Defaults to '/persistentvolumes'.
+
+|`openshift_provisioners_efs_name`
+| The `provisioner` name that _StorageClasses_ specify. Defaults to 'openshift.org/aws-efs'.
+
+|`openshift_provisioners_efs_nodeselector`
+| A map of labels (e.g. {"node":"infra","region":"west"} to select the nodes where the pod will land.
+
+|`openshift_provisioners_efs_supplementalgroup`
+| The supplemental group to give the pod in case it is needed for permission to write to the EFS file system. Defaults to '65534'.
+
+|===
+
+[[deploying-the-provisioners]]
+== Deploying the Provisioners
+
+You can deploy all provisioners at once or one at a time according to how you
+configure your OpenShift Ansible variables.
+
+The following examples show you how to deploy a given provisioner and then
+create and configure a corresponding xref:../install_config/persistent_storage/dynamically_provisioning_pvs.adoc[_StorageClass_].
+
+[[deploying-the-aws-efs-provisioner]]
+=== Deploying the AWS EFS Provisioner
+The following command sets the directory in the EFS volume to use to
+"/data/persistentvolumes", which must exist in the file system and be mountable
+and writeable by the provisioner pod.
+
+----
+$ ansible-playbook <OPENSHIFT_ANSIBLE_DIR>/common/openshift-cluster/openshift_provisioners.yml \
+   -e openshift_provisioners_install_provisioners=True \
+   -e openshift_provisioners_efs=True \
+   -e openshift_provisioners_efs_fsid=fs-47a2c22e \
+   -e openshift_provisioners_efs_region=us-west-2 \
+   -e openshift_provisioners_efs_aws_access_key_id=AKIAIOSFODNN7EXAMPLE \
+   -e openshift_provisioners_efs_aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY \
+   -e openshift_provisioners_efs_path=/data/persistentvolumes
```

Comment:
Has to be done manually... right now the ansible script does not even create a default storage class; but theoretically it could do that, create a pvc, create a test pod, and verify it works. This is a good idea to add for sure